### PR TITLE
Create 2015-10-09-react-in-a-state-of-confusion

### PR DESCRIPTION
**Pull request by lukehansell**
### Blog title
React, In a State of Confusion

### Blog description
Based on https://github.com/holidayextras/tripapplite/issues/3265#issue-110485005, a quick tutorial on maintaining state outside of React components.

#### Author
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 1
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 2
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

### CODE REVIEW

**Comment by robhuzzey** [comment]
Looks good to me, thanks :+1:

File: _posts/2015-10-09-react-in-a-state-of-confusion.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+But wait! Our component doesn't re-render when we click and our UI is out of date with our state! :astonished:
+
+
+To rectify this situation we'll put our rendering inside a function.
+
+    update = function() {
+      React.render(<Component clickCount={clickCount} onClickHandler={clickHandler} />, document.body)
+    }
+    update()
```
Comment (hxpaul) [inline _posts/2015-10-09-react-in-a-state-of-confusion.md]: got a mix of semicolons and not in here, also could put that as a self invoker `update = function( ) { }( );`

Comment (lukehansell) [inline _posts/2015-10-09-react-in-a-state-of-confusion.md]: I wanted to keep the examples as simple and obvious as possible. It's easier to miss the `()` on the end of the self invoker than it is to miss a call to `update()`.

Comment (shackpank) [inline _posts/2015-10-09-react-in-a-state-of-confusion.md]: It would also assign the return value of the function to `update`, rather than the function

**Comment by hxpaul** [comment]
:+1:

File: _posts/2015-10-09-react-in-a-state-of-confusion.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+layout: post
+title: React, in a state of confusion
+categories: programming
+author: luke_hansell
+tags: react programming state backbone
+comments: true
+excerpt: "Components should be stateless. Stateless where possible. Push state higher up the dependency tree. But what happens when you get to the top of the tree?"
+---
+
+We follow design principles with React to make them as composable and encapsulated as possible. To enable this we have a few guidelines which Facebook themselves promote:
```
Comment (Joezo) [inline _posts/2015-10-09-react-in-a-state-of-confusion.md]: Do we have a source for this?

**Pull request merged**